lib/client/index.js

```diff
@@ -682,11 +675,68 @@ function asyncIteratorFromResponse (response) {
   }
 }
 
+class Subscriptions {
+  constructor () {
+    /** @type {Map<string, {eventsource: EventSource, callbacks: Set<function>}>} */
+    this._subscriptions = new Map()
```

Comment:
Should we add some kind of maxSubscriptions value, to prevent memory leak. This is common pattern.

Reply:
Maybe, an LRU cache that closes evicted event sources makes sense.

There are two vectors here though, eventsource(s) and callback(s).

Don't have a strong opinion about this, except that it might be a hidden feature that surprises/confuses users. In general, we should encourage users to call the unsubscribe function and clean up after themselves.

But I will let Bitkit devs weigh in.

Reply:
Setting a max size of Map is enough and number should be some high number, maybe 50. the max size should be configurable, like how it works in Node.js event emitter.

Throw an error with descriptive error if they hit it

Reply:
https://nodejs.org/api/events.html#emittersetmaxlistenersn